### PR TITLE
Add libdicom

### PR DESCRIPTION
**Pull request by jcupitt**
I've left some test paths in for now, but it seems to work.

### CODE REVIEW

File: meson/meson.build
Comment (bgilbert) [inline meson/meson.build:13]: Hmm, it'd be nice if `check` was optional so we didn't need to handle it here.  We don't run tests anyway.  (I realize I'm the one that [made it mandatory](https://github.com/ImagingDataCommons/libdicom/pull/23)...)

File: build.sh
Comment (bgilbert) [inline build.sh:1]: Please rebase this on top of #77.  That PR is basically ready to merge (just need to look through it one more time) and cleans up a lot of stuff.

File: build.sh
Comment (bgilbert) [inline build.sh:24]: libdicom isn't a dependency of any of the other libraries, so I think it'd be clearer to put it at the end of the list.  Also we'll need to track `uthash` in this file, and `check` if we continue needing it.

File: build.sh
Comment (bgilbert) [inline build.sh:50]: This shouldn't be necessary.  For local testing you can use the [override directory](https://github.com/openslide/openslide-winbuild#substitute-sources), and I'm not so concerned about CI-testing this PR with a libdicom-enabled OpenSlide.

**Review by bgilbert — COMMENTED** [review]
Thanks!

File: meson/meson.build
Comment (bgilbert) [reply meson/meson.build:13]: PR in https://github.com/ImagingDataCommons/libdicom/pull/45.

File: build.sh
Comment (bgilbert) [reply build.sh:1]: Actually, wouldn't hurt to rebase on top of #81 either.  I expect that'll merge soon.

File: meson/meson.build
Comment (jcupitt) [reply meson/meson.build:13]: Nice! I merged this to main on my libdicom fork, I'll use that for this PR.

File: build.sh
Comment (jcupitt) [reply build.sh:50]: OK, reverted.

File: build.sh
Comment (jcupitt) [reply build.sh:1]: OK, moved on top of #77, testing now.

File: build.sh
Comment (bgilbert) [reply build.sh]: Sorry, I should have said: after the other dependencies but before OpenSlide.

Comment (jcupitt) [reply build.sh]: OK, done!

**Comment by bgilbert** [comment]
https://github.com/ImagingDataCommons/libdicom/pull/46 should fix the build failures.  Meanwhile, I've rebased and squashed the commits here and pushed [some fixes](https://github.com/openslide/openslide-winbuild/compare/0471b9550ab85f987e302faebd0c371020c598b3..85f36897b037545309561f890589d6d947d1873f).

For development purposes, the libdicom wrap here points to a Git branch.  This is okay for now, but will need to be fixed before we do any openslide-winbuild releases.  I've filed #82 to track this.

@jcupitt, PTAL and confirm that this version looks good to you.  Once I have your approval, and once the libdicom fixes land so CI can pass here, I'll merge.

**Comment by jcupitt** [comment]
If I run this locally I see:

```
[1246/1304] Linking target subprojects/libdicom/dcm-dump.exe
FAILED: subprojects/libdicom/dcm-dump.exe 
i686-w64-mingw32-gcc  -o subprojects/libdicom/dcm-dump.exe subprojects/libdicom/dcm-dump.exe.p/tools_dcm-dump.c.obj -Wl,--allow-shlib-undefined -static-libgcc -Wl,--enable-auto-image-base -Wl,--dynamicbase -Wl,--nxcompat -Wl,--start-group -lssp subprojects/libdicom/libdicom.a -Wl,--subsystem,console -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 -Wl,--end-group
/usr/lib/gcc/i686-w64-mingw32/12.2.1/../../../../i686-w64-mingw32/bin/ld: subprojects/libdicom/dcm-dump.exe.p/tools_dcm-dump.c.obj: in function `main':
/work/32/build/../../meson/subprojects/libdicom/tools/dcm-dump.c:22: undefined reference to `_imp__dcm_log_level'
```

And set of similar link errors. If look at libdicom.a I see:

```
$ nm libdicom.a | grep dcm_log_level
00000000 B _dcm_log_level
```

So I think libdicom is not being built exactly correctly. I'll try to find the issue.

**Comment by jcupitt** [comment]
Ah I see, it's just the missing `__declspec(dllimport)` in `dicom.h`. I'm not sure what the best fix here is.

**Comment by jcupitt** [comment]
It looks like `libdicom.wrap` needs a `wrap-file` section with a directory key so that `build.sh` can find the LICENCE file. I added a commit, perhaps it should be force pushed.

With this change, `-m32` and `-m64` bdist both work locally.

**Comment by bgilbert** [comment]
Whoops, good catch!  I had tested successfully with a libdicom override, and a build without the override that failed on the problems in https://github.com/ImagingDataCommons/libdicom/pull/46, but I hadn't tested with a successful non-override build.

**Review by bgilbert — APPROVED** [review]
I've squashed the second commit into the first one.

**Pull request merged**
